test/integration/infrastructure/infrastructure_test.go

```diff
@@ -0,0 +1,566 @@
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"path/filepath"
+	"time"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/ecs"
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/vpc"
+	"github.com/gardener/gardener-extension-provider-alicloud/pkg/alicloud"
+	. "github.com/gardener/gardener-extension-provider-alicloud/pkg/alicloud/matchers"
+	alicloudinstall "github.com/gardener/gardener-extension-provider-alicloud/pkg/apis/alicloud/install"
+	alicloudv1alpha1 "github.com/gardener/gardener-extension-provider-alicloud/pkg/apis/alicloud/v1alpha1"
+	"github.com/gardener/gardener-extension-provider-alicloud/pkg/controller/infrastructure"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	"github.com/gardener/gardener/pkg/operation/common"
+	"github.com/gardener/gardener/test/framework"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+const (
+	allCIDR     = "0.0.0.0/0"
+	vpcCIDR     = "10.250.0.0/16"
+	workersCIDR = "10.250.0.0/21"
+
+	secretName = "cloudprovider"
+
+	sshPublicKey       = "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQDSb1DJfupnWTfKJ0fmRGgnSx8A2/pRd5oC49qE1jFX+/J9L01jUyLc5sBKZXVkfU5q5h0JfbkhJXSIkzqE+rNPnJBI4e+8Lo2TVWLAvVZRA9Fg9Dk3mgkdVB+9qW2mIqtJF5GOWKuk7HkObwpY1pX8kHC/LJVfpNQpVBqWef0WJj6vbyjhlZ3vgRxK9I6wdJzjUYtNsDhvvBTy/IBg/xp82w9T2r3GVfnaTLMeQCW9mPviDKnQsrWMgVb2A0Z4c62EbzzLzQV4ScVJ6JMgOgkMqEPdbnKF8dEQcSu+/DQZoZt56Aeov7T4oamahj9/rIDX+WR1nOcfntIdhCyoB4lISkNFz/MlPC7O8HwJk4P7rojLGNk6xmn6NxY5CJGC2dVxFsb1bmm+fKHAp62mgwEoFZcDyIkcsmnmnID9u0rJNyMz84YUGZ/jEz8LePujDHcXiqgoLsKJ8gNRneISL9+m9s1VK7WxDDIbq8iWzR7XfAVE/GzKpVYkqrWCvjKEeFIDuDUnf3jghQCQMsXnJM7zGWr1tl+Dvl2Avxmj2xyUJXYHbXbl2aM434DgQySnV8JPzYH7EsTmvuhdb8SJIbb/NonFsSM+72HpSzVc083x4B++VL7oP1X8cly62pFVM1fi8sxBio48Hq5SmAUu9T4wUY4J+AKU6osFA/ATlMCIiQ== your_email@example.com"
+	sshPublicKeyDigest = "b9b39384513d9300374c98ccc8818a8b"
+)
+
+var (
+	accessKeyID     = flag.String("access-key-id", "", "Alicloud access key id")
+	accessKeySecret = flag.String("access-key-secret", "", "Alicloud access key secret")
+	region          = flag.String("region", "", "Alicloud region")
+	vpcID           = flag.String("vpc-id", "", "Alicloud existing VPC id")
+)
+
+func validateFlags() {
+	if len(*accessKeyID) == 0 {
+		panic("need an Alicloud access key id")
+	}
+	if len(*accessKeySecret) == 0 {
+		panic("need an Alicloud access key secret")
+	}
+	if len(*region) == 0 {
+		panic("need an Alicloud region")
+	}
+	if len(*vpcID) == 0 {
+		panic("need an Alicloud existing VPC id")
+	}
+}
+
+type aliClient struct {
```

Comment:
Currently it is not easy to leverage `pkg/alicloud/client.ClientFactory`, because functions `NewECSClient` and `NewVPCClient` return interfaces `ECS` and `VPC`, respectively, which are defined by ourselves. If we want to reuse these interfaces, we have to append and implement more functions which are already implemented by alicloud SDK. It is not a good design and will generate duplicated code. 

 I will follow up this problem in future PRs, doing some refactor and unify this part of code.